engine/baml-runtime/src/type_builder/mod.rs

```diff
@@ -244,6 +248,125 @@ impl TypeBuilder {
         Arc::clone(&self.recursive_type_aliases)
     }
 
+    pub fn add_baml(&self, baml: &str, rt: &crate::BamlRuntime) -> anyhow::Result<()> {
```

Comment:
This looks redundant with some other code in this PR. Good to factor out?

Mind adding docstring to this fuction? The name and the type signature don't make it obvious what it does.